src/containers/pages/InterventionPlan/UpdatePlan/CaseDetails/index.tsx

```diff
@@ -0,0 +1,164 @@
+  event: null,
+  eventId: null,
+  fetchEventsCreator: fetchEvents,
+  service: OpenSRPService,
+};
+
+/** async function to load a single event
+ * @param {UUID} eventId - id of the event to get
+ * @param {OpenSRPService} servant - the opensrp service
+ * @param {ActionCreator<FetchEventsAction>} actionCreator - the actionCreator
+ */
+const loadEvent = async (
+  eventId: UUID,
+  servant: typeof OpenSRPService,
+  actionCreator: ActionCreator<FetchEventsAction>
+) => {
+  const service = new servant(OPENSRP_FIND_EVENTS_ENDPOINT, OPENSRP_API_BASE_URL);
+  const filterParams = {
+    id: eventId,
+  };
+  service
+    .list(filterParams)
+    .then((response: RawEvent) => actionCreator([response]))
+    .catch((err: Error) => displayError(err));
+};
+
+/** CaseDetails component */
+export const CaseDetails: React.FC<CaseDetailsProps> = (props = defaultCaseDetailsProps) => {
+  const { eventId, service, event, fetchEventsCreator } = props;
+  const [isCollapsed, setCollapse] = React.useState<boolean>(true);
+  const toggleCollapse = () => setCollapse(!isCollapsed);
+
+  if (eventId === null) {
+    // if eventId is null return early
+    displayError(new Error(FAILED_TO_GET_EVENT_ID));
+    return null;
+  }
+
+  // make a call to the events endpoint and get event
+  useEffect(() => {
+    loadEvent(eventId, service, fetchEventsCreator).catch((err: Error) =>
+      growl(err.message, { type: toast.TYPE.ERROR })
+    );
+  }, []);
+
+  if (!event) {
+    return <Loading />;
+  }
+  const translatedEvent = translateEvent(event);
+  const { caseInformation, fociInformation, caseNumber, fociInformationTitle } = translatedEvent;
+
+  const mapObjectToElement = (obj: FlexObject) =>
+    Object.entries(obj).map(([property, value], index) => {
+      return (
+        <tr key={index}>
+          <td>{property}</td> <td>{value}</td>
+        </tr>
+      );
+    });
+
+  return (
+    <>
+      <h4>
+        {CASE_NUMBER}: {caseNumber}
+      </h4>
+      <div id="accordion">
+        <Card>
+          <CardHeader className={'pointer'} onClick={toggleCollapse}>
+            <h5>{CASE_DETAILS}</h5>
+          </CardHeader>
+          <Collapse isOpen={isCollapsed}>
+            <CardBody>
+              <h5 className="mb-3 bold">{CASE_INFORMATION}</h5>
+              <table className="table table-borderless">
+                {mapObjectToElement(caseInformation)}
+              </table>
+              <hr />
+              <h5 className="mb-3 bold"> {fociInformationTitle}</h5>
+              <table className="table table-borderless">
+                {mapObjectToElement(fociInformation)}
+              </table>
+            </CardBody>
+          </Collapse>
+        </Card>
+      </div>
+    </>
+  );
+};
+
+/** describes object returned by mapState */
+type MapStateToProps = Pick<CaseDetailsProps, 'event'>;
+/** describes mapDispatch */
+type DispatchToProps = Pick<CaseDetailsProps, 'fetchEventsCreator'>;
```

Comment:
Are these specific to this component? If not, should we move them to the `types` file?

Reply:
Yes this are specific to CaseDetails